Makefile-lib-llvm

```diff
@@ -33,7 +33,7 @@ pony_targets += stdlib test-stdlib stdlib-debug test-stdlib-debug test-examples
 
 .PHONY: $(pony_targets)
 $(pony_targets): $(llvm_config)
-	@PATH=$(new_path) $(MAKE) -f Makefile-ponyc config=$(config) $(MAKECMDGOALS) $(MAKEFLAGS)
+	@PATH=$(new_path) $(MAKE) -f Makefile-ponyc LLVM_VENDOR=true config=$(config) $(MAKECMDGOALS) $(MAKEFLAGS)
```

Comment:
[nit] `PATH` is set before `$(MAKE)`, I suppose we could set the new var there, too. OTOH,  `PATH` might affect `$(MAKE)`, so, there's a reason to treat it differently.

Reply:
[$(MAKE)](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html) is "magical" in gnu make so using it is unrelated to setting PATH. I set PATH so that when projects besides llvm are requested to be built by there presence in `llvm_project_list` such as clang and clang++ they are found before any system provided programs of the same name.

One thing I will do is expicitly set LLVM_CONFIG where appropriate. Setting PATH does mean `llvm-config` is found by Makefile-ponyc, but its better to be explicit.